key-manager/src/main/scala/org/bitcoins/keymanager/EncryptedMnemonic.scala

```diff
@@ -1,32 +1,60 @@
 package org.bitcoins.keymanager
 
-import java.time.Instant
-
 import org.bitcoins.core.compat.CompatEither
 import org.bitcoins.core.crypto._
 import org.bitcoins.crypto.{AesCrypt, AesEncryptedData, AesPassword, AesSalt}
 import scodec.bits.ByteVector
 
+import java.time.Instant
 import scala.util.{Failure, Success, Try}
 
-sealed trait MnemonicState {
+sealed trait SeedState {
   def creationTime: Instant
 }
 
+sealed trait DecryptedSeedState extends SeedState {
+  protected def strToEncrypt: String
+
+  def encrypt(password: AesPassword): EncryptedSeed = {
+    val Right(clearText) = ByteVector.encodeUtf8(strToEncrypt)
+    val (key, salt) = password.toKey
+
+    val encrypted = AesCrypt.encrypt(clearText, key)
+
+    EncryptedSeed(encrypted, salt, creationTime)
+  }
+}
+
 case class DecryptedMnemonic(mnemonicCode: MnemonicCode, creationTime: Instant)
```

Comment:
Should consider 


`private[this] val mnemonicCode: MnemonicCode, ...)`

Same with other instances of `xprv` in this file.

Reply:
We can't do this because then we can't access it for when we go to actually build the key manager

Reply:
Ok, well at minimum we should be able to do this? 

```scala
private[keymanager] val mnemonicCode: MenonicCode
private[keymanager] val xprv: ExtPrivateKey,
```

With this, only things inside of `org.bitcoins.keymanager._` can access these `val`s